code/components/citizen-scripting-v8/src/V8ScriptRuntime.cpp

```diff
@@ -1151,9 +1184,48 @@ result_t V8ScriptRuntime::Create(IScriptHost* scriptHost)
 	// set the 'window' variable to the global itself
 	context->Global()->Set(String::NewFromUtf8(GetV8Isolate(), "window", NewStringType::kNormal).ToLocalChecked(), context->Global());
 
+	std::string nativesBuild = "natives_21e43a33.js";
+
+	{
+		for (const auto& versionPair : g_scriptVersionPairs)
+		{
+			bool isGreater;
+
+			if (FX_SUCCEEDED(m_manifestHost->IsManifestVersionBetween(std::get<ManifestVersion>(versionPair).guid, guid_t{ 0 }, &isGreater)) && isGreater)
+			{
+				nativesBuild = std::get<const char*>(versionPair);
+			}
+		}
+	}
+
 	// run system scripts
 	result_t hr;
 
+	if (FX_FAILED(hr = LoadSystemFile(const_cast<char*>(va("citizen:/scripting/v8/%s", nativesBuild)))))

```

Comment:
It might more sense to make this a list of sequential loads, as it's ending up to be quite a few files.